### PR TITLE
externalize getVersion()

### PR DESCRIPTION
**Pull request by narcoticfresh**
We had a version determination in the MainController. this is clearly the wrong place..

So this contains
* A new CoreUtils service containing a getVersion() method - yes, using the same code as before
* Let MainController use that service

This is preparation so swagger and others can display the same version to the user..

### CODE REVIEW

File: src/Graviton/CoreBundle/Controller/MainController.php [hunk 1]
```diff
@@ -65,11 +65,7 @@ public function indexAction()
         );
 
         $response->headers->set('Link', (string) $links);
-
-        # @todo don't find the composer file like so, use packagist to find and parse it if possible
-        $composerFile = __DIR__.'/../../../../composer.json';
-        $composer = json_decode(file_get_contents($composerFile), true);
-        $response->headers->set('X-Version', $composer['version']);
+        $response->headers->set('X-Version', $this->container->get('graviton.core.utils')->getVersion());
```
Comment (lapistano) [inline src/Graviton/CoreBundle/Controller/MainController.php]: I propose to use a kernel response filter for that .. (I already implemented this in GRV-1347), But I guess for now core.utils is fine

**Comment by narcoticfresh** [comment]
what is a kernel response filter? can we merge this - as you said it's fine for now?

if you want a kernel response filter - whatever that is - you need to point me to some sources, not GRV's ;-)

**Comment by lapistano** [comment]
@narcoticfresh  I just created a PR for that: https://github.com/libgraviton/graviton/commit/50b495086f35dba3bae41aee40b5d6f762326773

**Comment by hairmare** [comment]
You also might want to call ``CoreUtils`` just ``VersionUtils`` or similar. Chances are we'll have more versiony stuff to do anyhow.

**Comment by lapistano** [comment]
@narcoticfresh as I said, I think it is fine for now. It will be moved to a kernel response filter once the other PR is merged ;)

**Comment by narcoticfresh** [comment]
@lapistano It's fine that a Listener provides the X-Version header - but the 'heavy lifting' *needs* to be done in a service so other parts (as swagger or other services) can also easily get the current version..

**Comment by hairmare** [comment]
Composer has the following on this issue.

> Optional if the package repository can infer the version from somewhere, such as the VCS tag name in > the VCS repository. In that case it is also recommended to omit it.

They then go on to say.

> Specifying the version yourself will most likely end up creating problems at some point due to human error.

Which is all very much true. I should now, I just broke #18 due to this.

Can we get the necessary info from ``composer.lock``? Should we grab it from git?

I've been looking for 3rd party solutions to this but seem to be running into a brick wall on every attempt.

**Comment by narcoticfresh** [comment]
Please.. This PR just wants to put the version determination into a own simple class - we don't need to solve the whole versioning problem here. Once externalized, we can always change the way the version is determined.

**Pull request merged**